rust/src/mqtt/mqtt_message.rs

```diff
@@ -0,0 +1,190 @@
+
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// written by Sascha Steinbiss <sascha@steinbiss.name>
+
+use crate::mqtt::mqtt_property::*;
+use crate::mqtt::parser::*;
+
+#[derive(Debug)]
+pub struct MQTTMessage {
+    pub header: FixedHeader,
+    pub op: MQTTOperation,
+}
+
+#[derive(Debug)]
+pub enum MQTTOperation {
+    UNASSIGNED,
+    CONNECT(MQTTConnectData),
+    CONNACK(MQTTConnackData),
+    PUBLISH(MQTTPublishData),
+    PUBACK(MQTTMessageIdOnly),
+    PUBREC(MQTTMessageIdOnly),
+    PUBREL(MQTTMessageIdOnly),
+    PUBCOMP(MQTTMessageIdOnly),
+    SUBSCRIBE(MQTTSubscribeData),
+    SUBACK(MQTTSubackData),
+    UNSUBSCRIBE(MQTTUnsubscribeData),
+    UNSUBACK(MQTTUnsubackData),
+    AUTH(MQTTAuthData),
+    PINGREQ,
+    PINGRESP,
+    DISCONNECT(MQTTDisconnectData),
+    TRUNCATED(MQTTTruncatedData),
```

Comment:
Could you add a comment to specify that `TRUNCATED` is special : not in the MQTT specification, but Suricate-Specific ?

Reply:
Of course.